### DISCUSSION TITLE
File copied from cache to workspace after dvc remove

### DISCUSSION THREAD

**Discussion by skrishnanv** [treeverse/dvc#7028] [General]
Hi all,
i was experimenting with the feature[https://dvc.org/doc/command-reference/add#example-transfer-to-the-cache]. the file name is data.xml
i am on WSL with the dvc project being somewhere in /mnt/c and the cache in /mnt/d/cache
I have enabled soft linking.
The download and linking was fine, i could see the link pointing to the cache.
Then i did dvc remove on the data.xml.dvc. Then i found that the original file is copied back to workspace but the symlink was gone. dvc gc -w removed the cache.
is this an expected behaviour?
Regards

**Comment by pmrowla** [comment]
Yes, this is expected behavior. `dvc remove` just tells DVC to stop tracking a file, it does not remove the actual data. To also remove the actual data you need to include the `--outs` option. From the `dvc remove` docs (https://dvc.org/doc/command-reference/remove#description):

> `dvc remove` doesn't remove files from the DVC cache or remote storage. Use `dvc gc` for that.

> Note that the actual output files or directories of the stage (outs field) are not removed by this command, unless the `--outs` option is used.

**Comment by skrishnanv** [comment]
Thanks,
I understood that part. I was wondering about the file popping back in the
workspace.. would there be a better place for popping it back. because if
this were a large file, I would see a 2gb file coming up. Just thinking out
aloud.
